modules/openapi-generator/src/main/java/org/openapitools/codegen/utils/ModelUtils.java

```diff
@@ -725,14 +725,33 @@ public static boolean isModel(Schema schema) {
         }
 
         // has properties
-        if (null != schema.getProperties()) {
+        if (null != schema.getProperties() && !schema.getProperties().isEmpty()) {
             return true;
         }
 
         // composed schema is a model, consider very simple ObjectSchema a model
         return schema instanceof ComposedSchema || schema instanceof ObjectSchema;
     }
 
+    /**
+     * Check to see if the schema is a model with properties only (non-composed model)
+     *
+     * @param schema potentially containing a '$ref'
+     * @return true if it's a model with at least one properties
+     */
+    public static boolean isModelWithPropertiesOnly(Schema schema) {
```

Comment:
How about naming this isModelWithPropertiesAndSetAddProps?
- Do you want this to be true for AnyType models with properties and additionalProperties set? Right now it is true for that case.
- Do you want this to be true for array/string/number models with properties and additionalProperties set? Right now it is true for that case.
How about limiting this to type object and AnyType models only?

Reply:
Thanks for the review.

I previously didn't check `getAdditionalProperties()` but got a few issues (change in samples). Adding the check seems to fix it but you're right that it needs to revised further.

I simply want to set the isModel flag correctly because now isMap is set to true for a simple model. I've not touched the code in `modules/openapi-generator/src/main/java/org/openapitools/codegen/IJsonSchemaValidationProperties.java` for a while so I welcome suggestions on other way to fix the problem.

Reply:
Let me look into what is happening with python experimental. I think it is changing because the component schema changed

Reply:
Root cause investigation is below in other comments

Reply:
Should this be true if the schema also has anyOf or allOf definitions?